pandas/core/generic.py

```diff
@@ -3630,6 +3630,93 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             return res
 
     # ----------------------------------------------------------------------
+    # Timeseries methods Methods
+
+    def asof(self, where, subset=None):
+        """
+        Return last good (non-NaN) value in object if subset is NaN for
+        requested date if the input is a Series
+
+        The last row without any NaN is taken (or the last row without
+        NaN considering only the subset of columns) if the input is a
+        DataFrame
+
+        .. versionadded:: 0.18.2 For DataFrame
+
+        If there is no good value, NaN is returned.
+
+        Parameters
+        ----------
+        where : date or array of dates
+        subset : string or list of strings, default None
+           if not None use these columns for NaN propogation
```

Comment:
propogation -> propagation. 

But maybe better: "use these columns to determine the last row without any NaN"